database/database_test.go

```diff
@@ -234,3 +234,180 @@ func TestStatesWithStatus(t *testing.T) {
 		}
 	}
 }
+
+// setTests is a struct that is used in testing SetIneligibleDomainStates
+var setTests = []struct {
+	description       string
+	domainStates      []IneligibleDomainState
+	wantStatusReports []string
+}{
+	{
+		"two domains",
+		[]IneligibleDomainState{
+			{Name: "youtube.com", Policy: "bulk-1-year"},
+			{Name: "garron.net", Policy: "bulk-1-year"},
+		},
+		[]string{"Updating 2 entries...", " done.\n"},
+	},
+	{
+		"bulk-18-week",
+		[]IneligibleDomainState{{Name: "gmail.com", Policy: "bulk-18-week"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+	{
+		"bulk-1-year",
+		[]IneligibleDomainState{{Name: "wikipedia.com", Policy: "bulk-1-year"}},
+		[]string{"Updating 1 entries...", " done.\n"},
+	},
+}
+
+// Test SetIneligibleDomainStates
+func TestSet(t *testing.T) {
```

Comment:
nit: Here and in other functions, it's common to name test functions TestNameOfFunctionTested, so this would become TestSetInelegibleDomains